docs/react-v9/contributing/rfcs/shared/build-system/stop-styles-transforms.md

```diff
@@ -0,0 +1,162 @@
+  }
+  .order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  🔴 We expect that a "div" below will have "padding-left: 5px", but instead, it has "padding: 10px".
+
+     The issue appears because the CDN bundle loads after the main bundle, resulting in style overrides. This occurs because ".order0" appears in both bundles, and CSS prioritizes the order of appearance. 
+  -->
+
+<div class="order0 order1"></div>
+```
+
+The single reliable solution is to prefix styles with a unique identifier. This prevents CSS rules from colliding, for example:
+
+```html
+<!-- PROPOSAL -->
+
+<!-- main bundle -->
+<style>
+  .main-order0 {
+    padding: 10px;
+  }
+  .main-order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  ✅ Now the "div" below will have "padding-left: 5px" as CSS rules don't clash anymore.
+  -->
+
+<div class="main-order0 main-order1"></div>
+```
+
+## Problem statement
+
+Integrating prefixing into Griffel is relatively straightforward. However, we currently undergo a pre-processing step in `@fluentui/react-components` that modifies JavaScript code.
+
+<details>
+  <summary>Original code</summary>
+
+```ts
+// packages/react-components/react-menu/src/components/MenuDivider/useMenuDividerStyles.styles.ts
+// 📝 output is simpfied
+
+const useStyles = makeStyles({
+  root: {
+    ...shorthands.margin('4px', '-5px', '4px', '-5px'),
+  },
+});
+```
+
+</details>
+
+<details>
+  <summary>Pre-processed code</summary>
+
+```ts
+// @fluentui/react-menu/lib/components/MenuDivider/useMenuDividerStyles.styles.js
+// 📝 output is simpfied
+
+const useStyles = /*#__PURE__*/ __styles(
+  {
+    root: {
+      B6of3ja: 'fvjh0tl',
+      t21cq0: ['f1rnx978', 'f1q7jvqi'],
+    },
+  },
+  {
+    d: ['.fvjh0tl{margin-top:4px;}', '.f1rnx978{margin-right:-5px;}', '.f1q7jvqi{margin-left:-5px;}'],
+  },
+);
+```
+
+</details>
+
+Addressing prefixing for pre-processed styles presents a challenge, as they have already been transformed into CSS rules and classes.
+
+## Proposal
+
+The long-term solution is to discontinue the pre-processing of styles in `@fluentui/react-components` and transfer this responsibility to the consuming application.
+
+> Note: Applications already undertake this responsibility, as first and third-party packages do not include pre-processed styles.
+
+### Option A: Stop pre-processing styles
+
+This option involves eliminating the pre-processing step from `@fluentui/react-components`, shifting the responsibility to the consuming application.
+
+This is the simplest option, albeit it necessitates adjustments in the consuming app and is not backward compatible. Consequently, apps not utilizing [AOT in Griffel](https://griffel.js.org/react/ahead-of-time-compilation/introduction) will experience noticeable slowdowns during initial loading.
```

Comment:
with this every consumer that wants to get AOT benefits would need to provide config like following correct for all node_modules using griffel, correct ?

```js
{
    test: /\.styles.js$/,
    enforce: 'pre',
    exclude: {
      and: [/node_modules/], // Exclude libraries in node_modules ...
      not: [
        // Except for a few of them that needs to be transpiled because they use modern syntax
        /@fluentui/[\\/]react-components/,
       // more patterns based on libraries used
      ]
    },
    use: {
      loader: '@griffel/webpack-loader',
      options: {}
    }
  }
```

It would be good to understand impact on slow down build times in bigger applications....

Reply:
This manual processing would also be needed for these UMD bundles you are referring right ? Which I'm wondering how feasible that might be if they are served from CDN....

Reply:
> with this every consumer that wants to get AOT benefits would need to provide config like following correct for all node_modules using griffel, correct ?

We **have to** the following already (https://github.com/microsoft/fluentui/pull/31134/files#r1577467675):

```js
module.exports = {
  module: {
    rules: [
      // node_modules anyway as they have both `makeStyles` & `__styles()` (after AOT)
      {
        test: /\.js$/,
        include: /node_modules/,
        enforce: "post",
        use: [
          { loader: GriffelCSSExtractionPlugin.loader },
          { loader: "@griffel/webpack-loader" },
        ],
      },
      // App files
      {
        test: /\.styles.ts$/,
        exclude: /node_modules/,
        enforce: "post",
        use: [
          { loader: GriffelCSSExtractionPlugin.loader },
          { loader: "@griffel/webpack-loader" },
        ],
      },
    ],
  },
};

```

> This manual processing would also be needed for these UMD bundles you are referring right ? Which I'm wondering how feasible that might be if they are served from CDN....

Expectation is that an app part that comes from CDN will be prefixed already and won't collide.